tools/chapel-py/chplcheck/rules.py

```diff
@@ -128,6 +128,10 @@ def MisleadingIndentation(context, root):
         for child in root:
             yield from MisleadingIndentation(context, child)
 
+            # dont call `.location` on comments
+            if isinstance(child, Comment) or isinstance(prev, Comment):
+                continue
```

Comment:
Maybe the better solution here is to set `prev` to the last non-comment (great?) grandchild instead of `list(grandchildren[-1])[0]`, and not to have the `isinstance(prev, Comment)` check here?

i.e. change

```Python
            if isinstance(child, Loop) and child.block_style() == "implicit":
                grandchildren = list(child)
                if len(grandchildren) > 0:
                    prev = list(grandchildren[-1])[0]
```

to

```Python
            prev = None # this is probably needed to reset `prev` between statements.
            if isinstance(child, Loop) and child.block_style() == "implicit":
                grandchildren = list(child)
                # I think it's fine to access [-1] on grandchildren without checking length --
                # what loop doesn't have a body?
                for blockchild in reversed(list(grandchildren[-1])):
                    if isinstance(blockchild, Comment):
                        continue
                    prev = blockchild
                    break
```

Reply:
Done, I needed to keep the check `if isinstance(child, Comment): continue`, which I moved to the top of the loop

Reply:
Yep, the `child` check is still needed. Thanks!